cmd/generate/generate.go

```diff
@@ -364,34 +354,26 @@ func main() {
 			log.Println("File ", projectFilePath, " not found")
 			metadata = nil
 		}
+		// Builder image might be provided without formatting '%s' string as plain value
+		builderImage := mustGatherBuilderImageFmt
+		if strings.Count(mustGatherBuilderImageFmt, "%s") == 1 {
```

Comment:
Cool. So this will generate the right Dockerfile for OpenShift CI where brew.registry.redhat.io is not available.
But we should still use the brew variant for Konflux build. We need to pass the build arg similar to other image overrides: https://github.com/openshift-knative/hack/blob/main/pkg/prowgen/prowgen_konflux.go#L100

Reply:
I believe we can just place it [here](https://github.com/openshift-knative/hack/blob/main/config/serverless-operator.yaml#L8) (and further below for release-1.35 branch) like this:
```
        - name: MUSTGATHER
          pullSpec: brew.registry.redhat.io/rh-osbs/openshift-ose-must-gather:latest
```

Reply:
:+1: 
Only one point from my side: `latest` seems to be pretty old (`2021-11-18T17:17:43.151324972Z`). I guess we have to use the correct tag here (e.g. `brew.registry.redhat.io/rh-osbs/openshift-ose-must-gather:v4.17`)

Reply:
For serverless 1.34 we were using this base image: `FROM openshift-ose-must-gather:v4.13.0 AS builder`. So, perhaps we want the oldest version from the list of supported ones.

Reply:
brew.registry.redhat.io/rh-osbs/openshift-ose-must-gather:v4.13 is upd-to-date: `"build-date": "2024-11-05T07:55:36"`